src/Control/Carrier/FossaApiClient/Internal/FossaAPIV1.hs

```diff
@@ -1596,3 +1599,44 @@ uploadReachabilityContent signedUrl bs = fossaReq $ do
         lbsResponse
         options
         (pure . requestEncoder)
+
+editProjectURLEndpoint :: Url 'Https -> Url 'Https
+editProjectURLEndpoint baseUrl = baseUrl /: "api" /: "cli" /: "project"
+
+editProject ::
+  (Has (Lift IO) sig m, Has Diagnostics sig m, Has Debug sig m) =>
+  ApiOpts ->
+  Text ->
+  ProjectMetadataRevision ->
+  m ProjectResponse
+editProject apiOpts projectLocator ProjectMetadataRevision{..} = fossaReq $ do
+  (baseUrl, baseOpts) <- useApiOpts apiOpts
+  let opts =
+        "projectLocator"
+          =: projectLocator
+          <> mkProjectOpts
+
+  res <- req PUT (editProjectURLEndpoint baseUrl) NoReqBody jsonResponse (baseOpts <> opts)
+  pure (responseBody res)
+  where
+    mkProjectOpts :: Option scheme
+    mkProjectOpts = mconcat totalOptions
```

Comment:
[nit] This name implies this is a function that can be used to do something, but it's actually just an `Option value`. Consider just calling it `projectOpts`.